ams/server/src/main/java/com/netease/arctic/server/optimizing/TaskRuntime.java

```diff
@@ -345,7 +347,10 @@ private Set<Status> getNext() {
     }
 
     public synchronized boolean tryAccepting(Status targetStatus) {
-      if (owner.isClosed() || !getNext().contains(targetStatus)) {
+      if (owner.isClosed() && targetStatus != Status.CANCELED) {
```

Comment:
I'm not sure if treating `CANCELED` as a special status here is a good approach, or if we should include `CANCELED` in the next status list for all possible statuses to express this logic.

Reply:
The owner here is the optimizing process. When I try to close the optimizing process and its tasks, the optimizing process is set to be `Closed` first, and then the tasks are rejected changing the status to `CANCELED`.

Reply:
I changed it to canceling all tasks before closing the optimizing process.